File: rye/src/main/java/com/rysonw/rye/frontend/token.java
package com.rysonw.rye.frontend;

public class Token {
    // Class fields, constructors, and methods go here
    final TokenType type;
    final String lex;
    final int line;

    // Example constructor
    public Token(TokenType type, String lex, int line) {
        this.type = type;
        this.lex = lex;
        this.line = line;
    }

    // Example getter
    public String getType() {
        return type;
    }

    // Example setter
    public void setType(String type) {
        this.type = type;
    }
}

File: rye/src/main/rye/frontend/Token.cs
using System;

namespace com.rysonw.rye.frontend
{
    public class Token
    {
        public TokenType Type { get; }
        public string Lexeme { get; }
        public object Literal { get; }
        public int Line { get; }

        public Token(TokenType type, string lexeme, object literal, int line)
        {
            Type = type;
            Lexeme = lexeme;
            Line = line;
            Literal = literal;
        }

        public override string ToString()
        {
            return $"{Type} {Lexeme} {Literal}";
        }
    }
}
